### PR TITLE
Copy operation state field from Application status

### PR DESCRIPTION
**Pull request by chetan-rns**
#### Description:
- Add a new field `operationState` to the status of GitOpsDeployment CR
- Add a new field to store the operation state in the ApplicationState DB table
- Copy the operation state from Application to GitOpsDeployment
- Compress/Decompress the operation state before accessing it from the table
- Add unit tests and e2e tests

#### Link to JIRA Story (if applicable):

https://issues.redhat.com/browse/GITOPSRVCE-619

### CODE REVIEW

**Comment by jparsai** [comment]
Hey @chetan-rns there are conflicts in PR, please resolve them, so that it is good for review.

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great, thanks @chetan-rns, and thanks for reviewing @jparsai! Some minor unit test changes that occurred to me during reviewing, then good to go.

File: backend-shared/db/types.go [hunk 1]
```diff
@@ -303,6 +303,8 @@ type ApplicationState struct {
 
 	Resources []byte `pg:"resources"`
 
+	OperationState []byte `pg:"operation_state"`
```
[the remark this answers is not in the document]
Comment (chetan-rns) [reply backend-shared/db/types.go:306]: Added the new field in db-migration test

**Comment by chetan-rns** [comment]
@jgwest @jparsai I've added the missing tests. Also, I moved some common functions (compress/decompress) to backend-shared to avoid duplication. PTAL.

**Review by jgwest — APPROVED** [review]
Looks great, thanks again @chetan-rns!

**Pull request merged**